### PR TITLE
Fix bug and addressed comments received.

### PR DESCRIPTION
**Pull request by mjethanandani**

### CODE REVIEW

File: src/yang/ietf-access-control-list.yang [hunk 1]
```diff
@@ -466,60 +466,71 @@ module ietf-access-control-list {
...
                The rules are selected based on the feature set
-               defined by the server and the acl-type defined.";
-
-            container eth {
-              when "derived-from(../../../../acl-type, " +
-                   "'acl:eth-acl-type')";
-              if-feature match-on-eth;
-              uses packet-fields:acl-eth-header-fields;
+               defined by the server and the acl-type defined.
+               If the packet does not match any of the fields
```
Comment (einarnn) [inline src/yang/ietf-access-control-list.yang]: Maybe reword to:

```
Any field not specified is effectively wild carded. Thus if no
matches are specified, it is the equivalent of 'any'."
```

I'll put this in my PR.

File: src/yang/ietf-access-control-list.yang [hunk 2]
```diff
@@ -436,7 +436,7 @@ module ietf-access-control-list {
            and value of this name, possibly space and special
            characters are not allowed.";
       }
-      leaf acl-type {
+      leaf type {
```
Comment (plajjan) [inline src/yang/ietf-access-control-list.yang:369]: I thought Martin's feedback was that we wanted to change acl-name to name but "type" was so generic that it was better to still keep "acl-type"!?

File: src/yang/ietf-access-control-list.yang [hunk 3]
```diff
@@ -325,38 +325,39 @@ module ietf-access-control-list {
...
+        leaf name {
           type leafref {
             path "../../../../../../acl/name";
           }
           description
-            "Reference to the ACL set name applied on ingress";
+            "Reference to the ACL name applied on ingress";
         }
 
         leaf type {
```
Comment (plajjan) [inline src/yang/ietf-access-control-list.yang]: This leaf needs to go away. It serves no purpose. acl/name is the key of the list so there is no point in having anything else for referencing an ACL.

File: src/yang/ietf-access-control-list.yang (hunk 2 above)
Comment (mjethanandani) [reply src/yang/ietf-access-control-list.yang:369]: Here is the e-mail from Martin:

I think 99% of the changes in this PR look good.  The one
exception is the typedef that used to be called "acl-type".  I think
it should still be called "acl-type".  "type" is too broad.  NOTE,
this is just the typedef; the leaf /access-lists/acl/type should keep
its name ("type").

File: src/yang/ietf-access-control-list.yang (hunk 3 above)
Comment (mjethanandani) [reply src/yang/ietf-access-control-list.yang]: By "this" you mean the "type" leaf - right? If so, I agree.

File: src/yang/ietf-access-control-list.yang (hunk 1 above)
Comment (mjethanandani) [reply src/yang/ietf-access-control-list.yang]: How about this?

If no matches are specified, it is the equivalent of 'any'.

File: src/yang/ietf-access-control-list.yang [hunk 4]
```diff
@@ -466,60 +466,71 @@ module ietf-access-control-list {
...
+                when "derived-from(../../../../type, " +
+                     "'acl:eth-acl-type')";
+                if-feature match-on-eth;
+                uses packet-fields:acl-eth-header-fields;
+                description
+                  "Rule set that matches ethernet headers.";
+              }
               description
-                "Rule set that supports ethernet headers.";
+                "Choice of layer 2 headers, currently only
```
Comment (plajjan) [inline src/yang/ietf-access-control-list.yang]: You will never be able to remove "currently only Ethernet" in the description. If an augmentation later comes in and adds something else, this description is plain wrong.

I'd suggest changing to: `Match layer 2 fields, for example Ethernet header fields`

File: src/yang/ietf-access-control-list.yang [hunk 5]
```diff
@@ -519,7 +519,8 @@ module ietf-access-control-list {
             if-feature "acl-aggregate-stats";
             config false;
             description
-              "Statistics gathered across all interfaces.";
+              "Statistics gathered acrross all attchment points for the
```
Comment (plajjan) [inline src/yang/ietf-access-control-list.yang]: Fix "acrross" and "attchment"

Comment (mjethanandani) [reply src/yang/ietf-access-control-list.yang]: Will fix. Thanks.
